gnosis/verification.k

```diff
@@ -79,13 +79,19 @@ module VERIFICATION
     rule nthbyteof(chop(#asWord(W : WS) &Int MASK), 0, N) => W &Int (MASK >>Byte #sizeWordStack(WS))
       requires #sizeWordStack(W : WS) <=Int 32
 
+    rule nthbyteof(_, _, _)  &Int (A => A &Int 255)
+      requires A >Int 255
+
     rule nthbyteof(WS, I, N) &Int 255 => nthbyteof(WS, I, N)
 
     rule nthbyteof(chop(MASK &Int #asWord(WS)) => chop(#asWord(WS) &Int MASK), _, _)
       requires #isConcrete(MASK)
       //version below doesn't work
       //requires notBool (getKLabel(MASK) ==K #klabel(`#asWord`))
 
+    // #asWord only keeps 32 bytes and chops the rest.
+    rule #asWord(W : WS => WS)
+      requires #sizeWordStack(W : WS) >Int 32
```

Comment:
have you seen this case actually happens?

Reply:
Yes. I've seen an #asWord with ~192 elements, The rules for `nthbyteof(#asWord() &Int MASK)` were applying and the spec was proved, although this was not correct.

Reply:
Would it be a potential bug of the program by any chance?

I mean, we don't know yet whether the original concrete #asWord definition is correct or not, in case that the argument is more than 32 bytes.  It is never tested in that case.

Could you elaborate a bit more in which context it happens?

Reply:
I didn't study it too much. Yi noticed that semantics of #asWord chops result to 32 byes. I added the right side condition to lemmas dealing with #asWord and got a failing spec because some #asWord had an argument longer than 32. Added this trimming lemma and all was good again.

We can possibly find the details in the jenkins log history.

Reply:
Best way to proceed: after this PR is merged, I'll delete the lemma and PR again, let's see what happens.

Reply:
sounds a good plan